src/Features/Core/Portable/Highlighting/Keywords/AbstractKeywordHighlighter.cs

```diff
@@ -74,13 +79,15 @@ private static bool AnyIntersects(int position, List<TextSpan> highlights)
     protected static TextSpan EmptySpan(int position)
         => new(position, 0);
 
-    internal static void AddTouchingTokens(SyntaxNode root, int position, List<SyntaxToken> tokens)
+    internal void AddTouchingTokens(SyntaxNode root, int position, ref TemporaryArray<SyntaxToken> tokens)
     {
-        AddTouchingTokens(root, position, tokens, findInsideTrivia: true);
-        AddTouchingTokens(root, position, tokens, findInsideTrivia: false);
+        AddTouchingTokens(root, position, ref tokens, findInsideTrivia: true);
+        if (_findInsideTrivia)
+            AddTouchingTokens(root, position, ref tokens, findInsideTrivia: false);
```

Comment:
no point going into trivia for most highlighters. can update others as appropraite.